detection_rules/version_lock.py

```diff
@@ -22,19 +22,64 @@
 ETC_DEPRECATED_RULES_PATH = Path(get_etc_path()) / ETC_DEPRECATED_RULES_FILE
 MIN_LOCK_VERSION_DEFAULT = Version("7.13.0")
 
+temp_lock = {}
+
 
 @dataclass(frozen=True)
 class VersionLockFileEntry(MarshmallowDataclassMixin):
     """Schema for a rule entry in the version lock."""
+
     rule_name: definitions.RuleName
     sha256: definitions.Sha256
     type: definitions.RuleType
     version: definitions.PositiveInteger
     min_stack_version: Optional[definitions.SemVer]
 
-    # TODO: need to exclude nested 'previous'
+    #  Skip validation on pre/post load/dump
     previous: Optional[Dict[definitions.SemVer, 'VersionLockFileEntry']]
 
+    @pre_load
+    def store_load_previous(self, data, **kwargs):
+        """Exclude and store nested 'previous' field prior to load"""
+        if "previous" in data:
+            temp_lock[data['rule_name']] = data['previous']
+            del data["previous"]
+        return data
+
+    @pre_dump
+    def store_dump_previous(self, data, **kwargs):
```

Comment:
Even though `Validation occurs on deserialization but not on serialization`, if we don't stash before dumping, invalid fields will not be added to the schema. We're instead explicitly ensuring no fields in `previous` are changed.